{code}
scala> def foo = {
     |   def bar : Nothing = {
     |     try {
     |  throw new RuntimeException 
     |     } catch {
     |         case _ => bar
     |    }
     |   }
     |   bar
     | }
foo: Nothing

scala> foo
java.lang.StackOverflowError
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9)
	at .bar$$1(<console>:9...
scala> 
{code}
